### PR TITLE
newman: init at 5.2.0

### PR DESCRIPTION
**Pull request by freezeboy**
###### Motivation for this change

Package new application

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by freezeboy** [comment]
inspiration from package spacegun: pkgs/applications/networking/cluster/spacegun

File: pkgs/development/web/newman/generate-dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -i bash -p nodePackages.node2nix
+
+node2nix \
+  --node-env ../../node-packages/node-env.nix \
```
Comment (doronbehar) [inline pkgs/development/web/newman/generate-dependencies.sh:5]: I know many standalone node packages use this, but I think it's not needed. My gut feeling tells me it should be a different node-env for each such package.

File: pkgs/development/web/newman/package.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[
+  { "newman": "git+https://github.com/postmanlabs/newman.git#v5.1.2" }
```
Comment (doronbehar) [inline pkgs/development/web/newman/package.json:2]: What's wrong with just `"newman"`? It will save you the trouble of manually updating it yourself.

Comment (freezeboy) [reply pkgs/development/web/newman/package.json:2]: I followed the spacegun package example in fact I'm ok to simplify it

File: pkgs/development/web/newman/generate-dependencies.sh (hunk 1 above)
Comment (freezeboy) [reply pkgs/development/web/newman/generate-dependencies.sh:5]: so just local to this folder ?

Comment (doronbehar) [reply pkgs/development/web/newman/generate-dependencies.sh:5]: yes.

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
Works as far as `--help` and `run --help` are concerned. Tested with `nixpkgs-review.

**Pull request merged**